repository_service_tuf_api/metadata.py

```diff
@@ -75,3 +76,129 @@ def post_metadata(payload: MetadataPostPayload) -> MetadataPostResponse:
     return MetadataPostResponse(
         data={"task_id": task_id}, message="Metadata update accepted."
     )
+
+
+class GetData(BaseModel):
+    metadata: Dict[Roles, TUFMetadata]
+
+
+class MetadataSignGetResponse(BaseModel):
+    data: Optional[GetData]
+    message: str
+
+    class Config:
+        with open("tests/data_examples/bootstrap/payload.json") as f:
+            content = f.read()
+        example = json.loads(content)
+        schema_extra = {
+            "example": {"metadata": {"root": example["metadata"]["root"]}}
+        }
+
+
+def get_metadata_sign() -> MetadataSignGetResponse:
+    bs_state = bootstrap_state()
+    # Adds support only when bootstrap is signing state
+    if bs_state.bootstrap is False and bs_state.state != "signing":
+        raise HTTPException(
+            status.HTTP_200_OK,
+            detail={
+                "message": "No signing available",
+                "error": (
+                    f"It requires bootstrap started. State: {bs_state.state}"
+                ),
+            },
+        )
+
+    settings_repository.reload()
```

Comment:
Ok, let me give some background here.
You are using the [`fresh()` which is the same as `get_fresh`](https://github.com/dynaconf/dynaconf/blob/e6914bfcaec6a6f956f6be357e7f6395304af2cb/dynaconf/base.py#L714). The difference is that I get the value at once.

What I do is using [`reload()`](https://github.com/dynaconf/dynaconf/blob/e6914bfcaec6a6f956f6be357e7f6395304af2cb/dynaconf/base.py#L1131) which recreates all the loaders for redis in Dynaconf, that is why is slow but is like forcing the recreation using Redis data.

The true is that I don't need it here in the because I'm not writing the redis in the same object, but adding it also doesn't bring issues.

We must have it in [`bootstrap_state`](https://github.com/repository-service-tuf/repository-service-tuf-api/blob/c9c9cf1a967fd36d6688cded1947279c525cb4aa/repository_service_tuf_api/__init__.py#L207)

The reason is during a request session for [bootstrap it writes](https://repository-service-tuf.readthedocs.io/en/stable/devel/design.html#rstuf-repository-settings-configuration) the [TUF Repository Settings](https://repository-service-tuf.readthedocs.io/en/stable/devel/design.html#tuf-repository-settings) `BOOTSTRAP` as `pre-<task-id>`.

If another session sends another request to add a artifact it will still see `BOOTSTRAP` as `None` if I use `get` or `refresh`, and when I `reload` it guarantee that all request sessions as the object recreated.

Reply:
That was useful info, but still, this did not give me an answer why `reload()` + `get()` will guarantee we get the latest values of all possible `SIGNING` variables in `Dynaconf`.
You yourself said that `reload` could be slow as it recreates all the loaders for redis in Dynaconf.

Reply:
It recreates reloading the information from Redis, which contains the data Dynaconf. Not sure I understood your doubt.